File: 1493. Longest Subarray of 1's After Deleting One Element/longestSubarray.py
from typing import List
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        left = 0
        right = 0
        zero = 0
        result = 0
        while right < len(nums):
            if nums[right] == 0:
                zero += 1
            while zero > 1:
                if nums[left] == 0:
                    zero -= 1
                left += 1
            result = max(result, right - left)
            right += 1
        
        return result

File: 1493. Longest Subarray of 1's After Deleting One Element/longestSubarray.java
class Solution {
    public int longestSubarray(int[] nums) {
        int left = 0;
        int right = 0;
        int zero = 0;
        int result = 0;
        while (right < nums.length) {
            if (nums[right] == 0) {
                zero++;
            }

            while (zero > 1) {
                if (nums[left] == 0) {
                    zero--;
                }
                left++;
            }

            result = Math.max(result, right - left);
            right++;
        }
        return result;
    }
}